src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/AsyncCausalityTracer.cs

```diff
@@ -0,0 +1,99 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace System.Threading.Tasks
+{
+    internal static class AsyncCausalityTracer
```

Comment:
Do we need the AsyncCausalityTracer type at all at this point?  Could call sites just be updated to use TplEventSource directly, as with all of the other call sites to it?

Reply:
We don't really need it since the WinRT dependent component from this class is removed - we could just update all the call sites to use TplEventSource directly. Would that be something you'd prefer over adding AsyncCausalityTracer back?

Reply:
>  Would that be something you'd prefer over adding AsyncCausalityTracer back?

Yes, that would be better - especially once the other cruft is deleted and these methods become trivial forwarders.

Reply:
Agreed.  We can look at `TplEventSource` if we want to understand the full list of events that can be emitted.  I'd only argue for keeping `AsyncCausalityTracer` if it were holding state, but that doesn't appear to be the case.